### ISSUE TITLE
CA1821 doesn't handle debug method call in debug context

### ISSUE THREAD

**Issue by Evangelink** [dotnet/roslyn-analyzers#3336]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.8 (Latest) and master

#### Diagnostic ID

Example: [CA1821](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1821)

#### Repro steps

Current unit test is broken as it gives the impression the code is handling the case where it is not.

Current test:

```csharp
public async Task CA1821CSharpTestRemoveEmptyFinalizersWithDebugFailAndDirective()
{
	await VerifyCS.VerifyAnalyzerAsync(@"
public class Class1
{
#if DEBUG
	// Violation will not occur because the finalizer will exist and 
	// contain code when the DEBUG directive is present. When the 
	// DEBUG directive is not present, the finalizer will not exist, 
	// and therefore not be empty.
	~Class1()
	{
		System.Diagnostics.Debug.Fail(""Finalizer called!"");
    }
#endif
}
");
}
```

Test as it should be:

```csharp
public async Task CA1821CSharpTestRemoveEmptyFinalizersWithDebugFailAndDirective()
{
	await new VerifyCS.Test
	{
		TestCode = @"
public class Class1
{
#if DEBUG
	// Violation will not occur because the finalizer will exist and 
	// contain code when the DEBUG directive is present. When the 
	// DEBUG directive is not present, the finalizer will not exist, 
	// and therefore not be empty.
	~Class1()
	{
		System.Diagnostics.Debug.Fail(""Finalizer called!"");
    }
#endif
}
",
		SolutionTransforms =
		{
			(solution, projectId) =>
			{
				var project = solution.GetProject(projectId);
				var parseOptions = ((CSharpParseOptions)project.ParseOptions).WithPreprocessorSymbols("DEBUG");

				return project.WithParseOptions(parseOptions)
					.Solution;
			},
		}
	}.RunAsync();
}
```

#### Expected behavior

No diagnostic is raised.

#### Actual behavior

A diagnostic is raised.

**Comment by Evangelink** [comment]
Relates to #1241 and https://github.com/dotnet/roslyn/issues/24569
